backend/app/libs/price_stats.py

```diff
@@ -37,6 +37,7 @@ def _make_returns_matrix(
             hist_prices.sort_index().loc[start:end].reset_index(),
         )
         for symbol, hist_prices in token_returns_dfs.items()
+        if not hist_prices.sort_index().loc[start:end].reset_index().empty
```

Comment:
It looks like `reset_index()` is useless.

Also, have you checked if `sort_index()` is really needed here? It's used at line 37 but that might be an unnecessary leftover from previous issues in the codebase.

Reply:
> Also, have you checked if sort_index() is really needed here? It's used at line 37 but that might be an unnecessary leftover from previous issues in the codebase.

Not necessarily an issue, but it suppresses a warning that might result in an error in later versions of pandas. Also, it's probably best to ensure that the timeseries is certainly monotonic before proceeding with applying the calculations.

> It looks like reset_index() is useless.

Removing this `reset_index` call causes this statement to error out
```python
returns_matrix = (
        reduce(
            reducer_on_symbol_and_hist_prices,
            hist_prices_items,
            hist_prices_items[0][1][["timestamp", "returns"]].rename(
                columns={"return": hist_prices_items[0][0]}
            ),
        )
        .drop("returns", axis=1)
        .set_index("timestamp")
    )
```
Specifically, on this line
```python
hist_prices_items[0][1][["timestamp", "returns"]].rename(
...
)
...
KeyError: "['timestamp'] not in index"
```

Reply:
My first remark was on line 40 only: as you are doing `.empty`, you very probably don't need to reset the index beforehand.

Let's not try improving this in this PR anyways, it's more of a general remark.

I was just concerned with making sure data structures are always using the same index throughout the code flow. Then if the reducers want to do specific stuff, they can still do it.

Reply:
Ah I see, in that case yes the `reset_index` used here is useless, will remove in my next commit.